lib/internal/source_map/prepare_stack_trace.js

```diff
@@ -17,6 +17,12 @@ const prepareStackTrace = (globalThis, error, trace) => {
     return f(error, trace);
   }
 
+  // `globalThis` is the global that contains the constructor which
```

Comment:
this doesn't match the normal behaviour exactly. Perhaps we should factor it into a common function that is called between them both?

The relevant behaviour is this:

https://github.com/nodejs/node/blob/e66c4deda8c63ca6c1557eed7c09ba01c6284e57/lib/internal/errors.js#L47-L67

Reply:
I was questioning whether we need the logic MainContextError.prepareStackTrace, I tested with express, and supporting the globalThis.Error addresses the reported issues I believe.

Reply:
It might not be express that depends on it, but stuff out there does, so removing it should be a separate process.

Reply:
@devsnek reasonable 👍 see update.